File: py-recitations-solutions/loops/square.py
# square.py

def getperfectsquares(n:int)->list:
    # Write a program that gets all the perfect squares between 1 and n. 
    # Store all these perfect squares in a list
    num = []

    for i in range(n):
        i = i + 1
        v = i**2

        if v > n:
            break
        elif v == n:
            num.append(v)
            break

        num.append(v)
    
    return num

if __name__ == "__main__":

    n = 1000

    # Output result to user
    psqrs = getperfectsquares(n)

    # excluding 1 and the last perfect square
    # psqrs = psqrs[1:len(psqrs)-1]

    print(f"There are {len(psqrs)} Perfect Squares between 1 and {n} :\n{psqrs}")

File: cpp-recitations-solutions/loops/square.cpp
#include <iostream>
#include <vector>
using namespace std;

vector<int> perfectsquares(int n){
  // function returns array of perfect squares between 1 and n (inclusive)
  // NOTE: n is a natural number greater than 1, duh
  vector<int> sq = {};
  int v;
  for(int i = 1; i < n; i++){
    v = i*i;
    if (v>n){
      break;
    }
    else if(v==n){
      sq.push_back(v);
      break;
    }
    sq.push_back(v);
  }

  return sq;
}

int main() {
  
  // Write a program that outputs all the perfect squares between 1 and 100
  int n = 100;

  vector<int> psqrs = perfectsquares(n);

  string output = "There are " + to_string(psqrs.size()) + " Perfect Squares between 1 and " + to_string(n) + " :\n ";

  cout << output;

  for (int i = 0; i < psqrs.size(); i++){
      if (i==0){
        cout<< "[";
      }
      cout << psqrs[i];
      if (i < psqrs.size()-1){
        cout<< ", ";
      }
      else if(i==psqrs.size()-1){
        cout<<"]";
      }
  }

}